backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/utils/CHExecUtil.scala

```diff
@@ -177,13 +177,14 @@ object CHExecUtil extends Logging {
       partitoining: HashPartitioning,
       childOutput: Seq[Attribute],
       output: Seq[Attribute]): NativePartitioning = {
-    val hashFields = partitoining.expressions.map {
-      case a: Attribute =>
-        BindReferences
-          .bindReference(a, childOutput)
-          .asInstanceOf[BoundReference]
-          .ordinal
-    }
+    val hashFields =
+      partitoining.expressions.map(ConverterUtils.getAttrFromExpr(_).toAttribute).map {
+        case a: Attribute =>
```

Comment:
Nit:
It looks unnecessary to keep `case a: Attribute =>`, as the first map's output is always `Attribute` type.

Reply:
fixed